astra/src/main/java/com/slack/astra/chunk/ReadOnlyChunkImpl.java

```diff
@@ -106,17 +149,19 @@ public ReadOnlyChunkImpl(
     this.snapshotMetadataStore = snapshotMetadataStore;
     this.searchMetadataStore = searchMetadataStore;
 
-    CacheSlotMetadata cacheSlotMetadata =
-        new CacheSlotMetadata(
-            slotId,
-            Metadata.CacheSlotMetadata.CacheSlotState.FREE,
-            "",
-            Instant.now().toEpochMilli(),
-            List.of(Metadata.IndexType.LOGS_LUCENE9),
-            searchContext.hostname,
-            replicaSet);
-    cacheSlotMetadataStore.createSync(cacheSlotMetadata);
-    cacheSlotMetadataStore.addListener(cacheSlotListener);
+    if (!Boolean.getBoolean("astra.ng.dynamicChunkSizes")) {
```

Comment:
Consider putting this string into a static final somewhere, and then using that throughout the code. This helps prevent any typos from slipping through unnoticed.